lib/engine/g_1856/corporation.rb

```diff
@@ -15,7 +16,9 @@ def initialize(game, sym:, name:, **opts)
         @game = game
         @started = false
         @escrow = nil
+        @log = @game.log
         super(sym: sym, name: name, **opts)
+        @all_shares = shares_by_corporation[self].dup
```

Comment:
Is there a better way?

Reply:
probably an indication that the corporation subclass is doing too much

Reply:
I've pulled out the logic from here. I still hope there's a better way to get access to the shares in the corporation so that they can be modified (change percentage & cert_size) for share issuing and the super weird handling of the case of 1-share CGR presidency